### PR TITLE
ENT-1463: Prepare node-api for determination.

### PR DESCRIPTION
**Pull request by chrisr3**
Refactor some of the serialisation code so that we can remove non-deterministic elements.

### CODE REVIEW

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/amqp/SerializerFactory.kt [hunk 1]
```diff
@@ -247,7 +246,7 @@ open class SerializerFactory(
                 // preserve the actual message locally
                 loggerFor<SerializerFactory>().apply {
                     error("${e.message} [hint: enable trace debugging for the stack trace]")
-                    trace(e.getStackTraceAsString())
+                    trace("", e)
```
Comment (exFalso) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/amqp/SerializerFactory.kt]: is this correct? @fenryka

File: build.gradle [hunk 2]
```diff
@@ -21,7 +21,7 @@ buildscript {
     // TODO: Upgrade gradle-capsule-plugin to a version with capsule:1.0.3
     ext.capsule_version = '1.0.1'
 
-    ext.asm_version = '0.5.3'
```
Comment (fenryka) [inline build.gradle]: That's a hell of a jump

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/amqp/SerializerFactory.kt (hunk 1 above)
Comment (fenryka) [reply node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/amqp/SerializerFactory.kt]: If it pretty prints the exception into the log it's fine - @chrisr3  is that the intnetnio?

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/carpenter/ClassCarpenter.kt [hunk 3]
```diff
@@ -155,7 +158,7 @@ class ClassCarpenter(cl: ClassLoader = Thread.currentThread().contextClassLoader
```
Comment (fenryka) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/carpenter/ClassCarpenter.kt]: remove the jvm extension func on string as it's been replaced

File: build.gradle (hunk 2 above)
Comment (chrisr3) [reply build.gradle]: This version number was never used by Corda, but it is _now_. The version number itself hasn't really changed at all - Kryo was always implicitly using ASM 5.0.4.

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/carpenter/ClassCarpenter.kt (hunk 3 above)
Comment (chrisr3) [reply node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/carpenter/ClassCarpenter.kt]: It has _one_ remaining user; but perhaps I can get rid of that too?

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/amqp/SerializerFactory.kt (hunk 1 above)
Comment (chrisr3) [reply node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/amqp/SerializerFactory.kt]: Yes. And removing the `getStackTraceAsString()` reference from `node-api-deterministic`.

**Pull request merged**